iothub_client/samples/iothub_ll_client_x509_sample/readme.md

```diff
@@ -0,0 +1,54 @@
+# Azure IoT Hub Client Sample: Using X509 Client Certificates
+
+## Build instructions
+
+1. Modify `.\iothub_ll_client_x509_sample.c`: update `connectionString`, `x509certificate` and 
+`x509privatekey`.
+   (Optionally, you can change configuration regarding transport, global endpoint, etc.)
+
+1. If you are using an OpenSSL engine, uncomment the `#define SAMPLE_OPENSSL_ENGINE` and set it to the
+   OpenSSL Engine name. Instead of a PEM certificate, `x509privatekey` will contain the private key ID
+   as expected by the OpenSSL engine. (e.g. in case of engine `pkcs11`, `x509privatekey` will contain
+   a PKCS#11 URI such as `pkcs11:object=test-privkey;type=private?pin-value=1234`.)
+
+1. Configure and build the SDK and sample
+
+    ```sh
+    # At the root of the repository
+    cd build
+    cmake ..
+    cmake --build . -j
+    ```
+
+## Run the sample
+
+```sh
+# At the root of the repository
+cd build
+./iothub_client/samples/iothub_ll_client_x509_sample/iothub_ll_client_x509_sample
+```
+
+## Configuring the TLS Stack
+
+### TLS Options
+
+Additional TLS stack specific configuration is available through `IoTHubDeviceClient_LL_SetOption`.
+
+### Long X509 Client Certificate Chains
+
+When using chains with more than one level, the device must send all chain certificates up to but 
+not including the CA configured on the service. The certificates will be sent part of the TLS 
+handshake in order for the service to build a valid chain. 
+
+For TLS stacks such as OpenSSL and mbedTLS, this is achieved by creating a PEM certificate store 
+(i.e. concatenating the Intermediates as well as the Device Certificate PEM contents) and passing 
+it to the C-SDK via `x509certificate`.
```

Comment:
is this via the option or via the variable? we should be specific.

Reply:
Good point: this is via the variable. Adding this to the doc here as well as at the top.